internal/core/command/init.go

```diff
@@ -27,19 +27,24 @@ import (
 	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
 	"github.com/edgexfoundry/go-mod-core-contracts/clients/metadata"
 	"github.com/edgexfoundry/go-mod-core-contracts/clients/types"
+
 	registryTypes "github.com/edgexfoundry/go-mod-registry/pkg/types"
 	"github.com/edgexfoundry/go-mod-registry/registry"
 
 	"github.com/edgexfoundry/edgex-go/internal"
+	"github.com/edgexfoundry/edgex-go/internal/core/command/interfaces"
 	"github.com/edgexfoundry/edgex-go/internal/pkg/config"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/db"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/db/mongo"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/db/redis"
 	"github.com/edgexfoundry/edgex-go/internal/pkg/startup"
 	"github.com/edgexfoundry/edgex-go/internal/pkg/telemetry"
 )
 
 var Configuration *ConfigurationStruct
 var LoggingClient logger.LoggingClient
 var mdc metadata.DeviceClient
```

Comment:
Should we replace the above client with additional DBClient functions?

Reply:
what do you mean...  to remove _**var mdc metadata.DeviceClient**_  and replace all calls (getAllDevices, getDeviceById, getDeviceByName) made from Command service to Metadata service  with new functions in the local DBClient ?

Reply:
Yes. I'm asking what you think of doing that. On the one hand, I would like to do it to eliminate the cross-service hop. On the other hand, metadata is authoritative for device information so the call makes sense. I'm conflicted. Asking for your opinion.

Reply:
I am conflicted as well, but ... by that moment my answer will be -  just for the sake of micro service architecture it is good to keep the cross-service calls. 
(nevertheless that with this PR the micro-service architecture pillars between core-command and metadata services will be broken.) 
On the other hand should  we blindly follow the the best-practice by providing own database per each service? Does it mean/is this enough condition to say that two service are really decoupled? (I still haven't get to the end of DDD book you recommenced )

Reply:
OK, we'll leave this as is for now.